courses/sync_external_courses/external_course_sync_api.py

```diff
@@ -376,46 +377,48 @@ def update_external_course_runs(external_courses, keymap):  # noqa: C901, PLR091
             course_run, course_run_created, course_run_updated = (
                 create_or_update_external_course_run(course, external_course)
             )
+            if course_run:
+                if course_run_created:
+                    stats["course_runs_created"].add(course_run.external_course_run_id)
+                    log.info(
+                        f"Created Course Run, title: {external_course.course_title}, external_course_run_id: {course_run.external_course_run_id}"  # noqa: G004
+                    )
+                elif course_run_updated:
+                    stats["course_runs_updated"].add(course_run.external_course_run_id)
+                    log.info(
+                        f"Updated Course Run, title: {external_course.course_title}, external_course_run_id: {course_run.external_course_run_id}"  # noqa: G004
+                    )
 
-            if course_run_created:
-                stats["course_runs_created"].add(course_run.external_course_run_id)
                 log.info(
-                    f"Created Course Run, title: {external_course.course_title}, external_course_run_id: {course_run.external_course_run_id}"  # noqa: G004
-                )
-            elif course_run_updated:
-                stats["course_runs_updated"].add(course_run.external_course_run_id)
-                log.info(
-                    f"Updated Course Run, title: {external_course.course_title}, external_course_run_id: {course_run.external_course_run_id}"  # noqa: G004
+                    f"Creating or Updating Product and Product Version, course run courseware_id: {course_run.external_course_run_id}, Price: {external_course.price}"  # noqa: G004
                 )
 
-            log.info(
-                f"Creating or Updating Product and Product Version, course run courseware_id: {course_run.external_course_run_id}, Price: {external_course.price}"  # noqa: G004
-            )
-
-            if external_course.price:
-                product_created, product_version_created = (
-                    create_or_update_product_and_product_version(
-                        external_course, course_run
+                if external_course.price:
+                    product_created, product_version_created = (
+                        create_or_update_product_and_product_version(
+                            external_course, course_run
+                        )
                     )
-                )
-                if product_created:
-                    stats["products_created"].add(course_run.external_course_run_id)
+                    if product_created:
```

Comment:
nit: may be we can move this into separate functions to reduce nested blocks in single function